connectionpool_systems_test.go

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) 2015 The gocql Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+// +build conn_pool
+
+package gocql
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+)
+
+// connection pool behavior test when nodes are removed from the cluster
+// to run this test, see connectionpool_systems_test.sh
+
+var (
```

Comment:
When I created this test the idea was that I wanted to just test the behavior of NewSimplePool, NewRoundRobinConnPool and NewTokenAwareConnPool when the ccm cluster was changed by adding or removing a number of nodes. The go test itself is pretty sloppy since I was exploring situations where policyConnPool performed sub par compared to SimplePool and I wasn't sure what exactly par was until I found situations where SimplePool actually returns nil connections and policyConnPool doesn't.

So the problems with this test that make it less than ideal to merge into integration.sh:
- SimplePool has a race condition where occasionally nil will be returned by Pick.
- The test actually makes calls to ccm unlike the tests in cassandra_test.go and I'm sure in some cases the cluster may be left in an changed state (e.g. nodes stopped).

I'm totally open to moving these tests into the integration though. These tests though would need to be written a bit cleaner and with more time efficiency. Also the race condition in SimplePool which leads to nil being returned by Pick would need to be resolved (or SimplePool should just be removed and replaced with NewRoundRobinConnPool)